### PR TITLE
Javadoc workflow in GH actions

### PR DESCRIPTION
**Pull request by qsdigor**
Workflow or GH action to generate JavaDoc and upload it to S3.

### CODE REVIEW

**Comment by qsdigor** [comment]
The current java version set is 8. It can be increased to 11 or 17 if required.

File: .github/workflows/javadoc.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
```
Comment (QuintinWillison) [inline .github/workflows/javadoc.yml:12]: This line is not required, as it's the default.

File: .github/workflows/javadoc.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: eu-west-2
+          role-to-assume: arn:aws:iam::${{ secrets.ABLY_AWS_ACCOUNT_ID_SDK }}:role/ably-sdk-builds-ably-java
+          role-session-name: "${{ github.run_id }}-${{ github.run_number }}"
+
+      - name: Set up the Java JDK
```
Comment (QuintinWillison) [inline .github/workflows/javadoc.yml:24]: The 'J' in JDK stands for 'Java'. But, arguably, the action name is self-explanatory so a `name` field for this step is superfluous. I would remove it.

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
Just a couple of minor semantic tweaks, please.

File: .github/workflows/javadoc.yml (hunk 2 above)
Comment (qsdigor) [reply .github/workflows/javadoc.yml:24]: I agree. I will remove it.

File: .github/workflows/javadoc.yml (hunk 1 above)
Comment (qsdigor) [reply .github/workflows/javadoc.yml]: Agreed, I removed it. [link](https://github.com/ably/ably-java/pull/832/files/6a9746d2ab467f28d88e3d1bac496dd055a44b9c..8f7b0ece28742fd7c86dcf9ac383c933063a4c7f)

**Review by ikbalkaya — APPROVED** [review]
LGTM

**Review by QuintinWillison — APPROVED** [review]

**Review by KacperKluka — APPROVED** [review]
LGTM 👍

**Pull request merged**